src/baldr/graphtile.cc

```diff
@@ -214,19 +221,15 @@ graph_tile_ptr GraphTile::CacheTileURL(const std::string& tile_url,
     return nullptr;
   }
 
-  auto uri = MakeSingleTileUrl(tile_url, graphid);
-  auto result = tile_getter->get(uri);
+  auto fname = valhalla::baldr::GraphTile::FileSuffix(graphid.Tile_Base(),
+                                                      valhalla::baldr::SUFFIX_NON_COMPRESSED, false);
+  tile_getter->set_path_pattern(valhalla::baldr::GraphTile::kTilePathPattern);
```

Comment:
do we need to do this every time? why cant we do this once in the constructor?

Reply:
right, my bad

Reply:
at the end, i think i will disagree with this as CacheTileURL method (static method, not sure it is good to change this for now) receives tile_getter as a parameter and only before trying to load tiles from remote storage inside CacheTileURL we know that we have to look for kTilePathPattern. I mean GraphTile constructor knows nothing about tile_getter, GraphReader knows nothing about GraphTile implementation and tile_getter_t/curl_tile_getter_t can be used with different use cases where we don't need to search for kTilePathPattern.

Reply:
I mean it should go right here:
https://github.com/valhalla/valhalla/blob/master/src/baldr/graphreader.cc#L485
In the tile getter constructor

Reply:
Probably the real question I should have asked is why put the url pattern into tile getter at all?

Reply:
> I mean it should go right here: https://github.com/valhalla/valhalla/blob/master/src/baldr/graphreader.cc#L485 In the tile getter constructor

yeah, i understood, that is why i wrote that  "GraphReader knows nothing about GraphTile implementation". it just seems for me that this is work for the current GraphTile implementation.